src/Mvc/Mvc.RazorPages/src/DependencyInjection/PageConventionCollectionExtensions.cs

```diff
@@ -378,8 +446,18 @@ public static PageConventionCollection AuthorizeAreaFolder(this PageConventionCo
                 throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(folderPath));
             }
 
-            var authorizeFilter = new AuthorizeFilter(policy);
-            conventions.AddAreaFolderApplicationModelConvention(areaName, folderPath, model => model.Filters.Add(authorizeFilter));
+            var authorizeAttribute = new AuthorizeAttribute(policy);
```

Comment:
is this attribute instance used anywhere? the endpoint metadata is newing another one up below